mypy.ini

```diff
@@ -18,6 +18,9 @@ check_untyped_defs = True
 [mypy-scipy.spatial.tests.test_spherical_voronoi]
 check_untyped_defs = True
 
+[mypy-scipy.spatial.tests.test_distance]
+ignore_errors = True
+
```

Comment:
```suggestion
```

Hello, @rgommers and @BvB93. The CI job failed as this was the overlapping change in gh-13613. That's what `mypy` reports in the first line:

```
/home/runner/work/scipy/scipy/mypy.ini: While reading from '/home/runner/work/scipy/scipy/mypy.ini' [line 385]: section 'mypy-scipy.spatial.tests.test_distance' already exists
```

I think this made mypy ignore the `mypy.ini` config file and led to a deluge of errors. To confirm this was indeed the case, I removed these lines from the changes in this PR and everything passed again. Can you please confirm it @rgommers or @BvB93?

PS: Sorry for the inconvenience that this may have cost!

Reply:
It looked to me like that was only a small part of the problem, but I didn't confirm. Let me try to reproduce locally.

Did the new CI job pass for you on master of your own fork?

Reply:
I have pushed it on my master. I will do that and let you know.

Reply:
Mypy passes on my master: https://github.com/tirthasheshpatel/scipy/runs/1997516154

Reply:
Thanks for checking! I've now verified it locally too, and with gh-13622 a simple `mypy scipy` should also pass (avoids the long builds, I'm not a `runtests.py` user).

I'll revert my revert again.